src/tribler/core/components/libtorrent/torrentdef.py

```diff
@@ -135,6 +136,24 @@ async def load_from_url(url):
             body = await response.read()
         return TorrentDef.load_from_memory(body)
 
+    def _filter_characters(self, name: bytes) -> str:
+        """Sanitize the names in path to unicode by replacing out all
+        characters that may -even remotely- cause problems with the '?'
+        character.
+
+        :param name: the name to sanitize
+        :type name: bytes
+        :return: the sanitized string
+        :rtype: str
+        """
+        def filter_character(char: int) -> str:
+            if 0 < char < 128:
+                return chr(char)
+            self._logger.debug("Bad character 0x%X", char)
+            return "?"
+
+        return "".join(map(filter_character, name))
```

Comment:
I see that this is a slightly modified version of the original code. Feel free to add a test for it if you have some spare time. If not, I'll handle it for this PR.

Reply:
The function was repeated twice. Because of this I extracted it to an underscore-prefixed method, along with the obvious change of removing the list comprehension from `str.join`. I'll try to add a test for it.